pkg/manager/task_migrate_hash.go

```diff
@@ -0,0 +1,86 @@
+package manager
+
+import (
+	"os"
+	"path/filepath"
+	"sync"
+
+	"github.com/stashapp/stash/pkg/logger"
+	"github.com/stashapp/stash/pkg/models"
+	"github.com/stashapp/stash/pkg/utils"
+)
+
+// MigrateHashTask renames generated files between oshash and MD5 based on the
+// value of the useMD5 flag.
+type MigrateHashTask struct {
+	Scene  *models.Scene
+	useMD5 bool
+}
+
+// Start starts the task.
+func (t *MigrateHashTask) Start(wg *sync.WaitGroup) {
+	defer wg.Done()
+
+	if !t.Scene.OSHash.Valid || !t.Scene.Checksum.Valid {
+		// nothing to do
+		return
+	}
+
+	oshash := t.Scene.OSHash.String
+	checksum := t.Scene.Checksum.String
+
+	oldHash := oshash
+	newHash := checksum
+	if !t.useMD5 {
+		oldHash = checksum
+		newHash = oshash
+	}
+
+	oldPath := filepath.Join(instance.Paths.Generated.Markers, oldHash)
+	newPath := filepath.Join(instance.Paths.Generated.Markers, newHash)
+	t.migrate(oldPath, newPath)
+
+	scenePaths := GetInstance().Paths.Scene
+	oldPath = scenePaths.GetThumbnailScreenshotPath(oldHash)
+	newPath = scenePaths.GetThumbnailScreenshotPath(newHash)
+	t.migrate(oldPath, newPath)
+
+	oldPath = scenePaths.GetScreenshotPath(oldHash)
+	newPath = scenePaths.GetScreenshotPath(newHash)
+	t.migrate(oldPath, newPath)
+
+	oldPath = scenePaths.GetStreamPreviewPath(oldHash)
+	newPath = scenePaths.GetStreamPreviewPath(newHash)
+	t.migrate(oldPath, newPath)
+
+	oldPath = scenePaths.GetStreamPreviewImagePath(oldHash)
+	newPath = scenePaths.GetStreamPreviewImagePath(newHash)
+	t.migrate(oldPath, newPath)
+
+	oldPath = scenePaths.GetTranscodePath(oldHash)
+	newPath = scenePaths.GetTranscodePath(newHash)
+	t.migrate(oldPath, newPath)
+
+	oldPath = scenePaths.GetSpriteVttFilePath(oldHash)
+	newPath = scenePaths.GetSpriteVttFilePath(newHash)
+	t.migrate(oldPath, newPath)
+
+	oldPath = scenePaths.GetSpriteImageFilePath(oldHash)
+	newPath = scenePaths.GetSpriteImageFilePath(newHash)
+	t.migrate(oldPath, newPath)
+}
+
+func (t *MigrateHashTask) migrate(oldName, newName string) {
+	oldExists, err := utils.FileExists(oldName)
+	if err != nil {
+		logger.Errorf("Error checking existence of %s: %s", oldName, err.Error())
```

Comment:
should we switch from error level to debug or warning here?
It pollutes the log with `Error checking existence of ...` for marker,webp and transcoded files ( for the majority of my scenes i don't have those generated). The specific files not existing shouldn't be an error

Reply:
Should've only been logging due to other error. Fixed.